components/renku_data_services/secrets/core.py

```diff
@@ -44,10 +44,19 @@ async def create_k8s_secret(
     if len(missing_secret_ids) > 0:
         raise errors.MissingResourceError(message=f"Couldn't find secrets with ids {', '.join(missing_secret_ids)}")
 
-    if key_mapping:
-        if set(key_mapping) != requested_secret_ids:
+    def ensure_list(value: str | list[str]) -> list[str]:
+        return [value] if isinstance(value, str) else value
+
+    key_mapping_list = {key: ensure_list(key_mapping[key]) for key in key_mapping} if key_mapping else None
```

Comment:
nitpick: it threw me off a bit at first glance that this is called `key_mapping_list` when it's a dict

Reply:
Do you have a suggestion? The variable is `key_mapping` except we ensure all values are `list[str]`.

Reply:
not really, maybe something like aggregated_key_mappings